### ISSUE TITLE
Install GitHub webhook on all repos

### ISSUE THREAD

**Issue by chadwhitacre** [getsentry/eng-pipes#193]
GitHub provides [three kinds of webhooks](https://docs.github.com/en/developers/webhooks-and-events/webhooks/about-webhooks):

1. repo
1. org
1. App

Right now we have a repo-level webhook on four repos—`getsentry`, `onpremise`, `sentry`, and `sentry-docs`— and in order to support our metrics and automation goals we want to install the webhook globally for all repos. We want to use an App-level webhook for this rather than an org-level webhook, so that we can delegate webhook management without oversharing org admin. The challenge is to do this without dropping or doubling events.

### Parameters

1. We log GitHub events from the webhook into a BigQuery table, and we query that table for our OSS SLOs.
1. BigQuery is designed to be append only. The best practice is to deduplicate in the application layer (if dupes are expected long-term), or rebuild the table (as a one-off).
    1. We have short-term dupes, so we don't to rewrite all our queries, which would be cumbersome and error-prone, easy to miss here or there, but ...
    1. Rebuilding the table is too coarse-grained for what we want. We can tolerate _some_ temporary corruption in our metrics, but rebuilding the table is too heavy to do often enough to keep metrics corruption within tolerances (< 1 minute?). We also have existing dupes in the table, and I'm not sure what removing those would do.
1. BigQuery _does_ support deletion, though. It's just kind of a second-class citizen.
1. We are only getting a few hundred new records per day in `github_events`.
1. We don't need to generalize or overthink this, we can handle this as a one-off with temporary fields/code that we remove after the webhooks are cut over.

#### tl;dr

Let's not overthink this. Our volume is so low that we should just minimize the race condition by clicking quickly. Especially if we time it for a slow time of day/week we will most likely not have any dupes. At most we should expect < 10 dupes, and we can identify and clean those up manually.

### Plan

1. Set up an App with a deactivated(!) webhook (deets below) that I can manage.
1. Temporarily grant me admin on `sentry` and `getsentry` (I already have it on `onpremise` and `sentry-docs`). 
    1. Plan B: Sync with Ops, have them click buttons.
1. Temporarily grant me updateData perms on `github_events`.
    1. Plan B: Sync with Data, have them dedupe if it comes to that.
1. Quickly! ... activate the App webhook and deactivate the repo webhooks.
    1. `getsentry`
    1. `sentry`
    1. `sentry-docs`
    1. `onpremise`
1. Remove temporary perms (if we went that route).
1. Manually identify, delete, and reinsert any duplicates via the BigQuery console.

### Webhook Details

1. **Payload URL:** https://product-eng-webhooks-vmrqv3f7nq-uw.a.run.app/metrics/github/webhook
1. **Content type:** application/json
1. **Secret:** [yes]
1. **SSL verification:** Enable SSL verification
1. **Which events would you like to trigger this webhook?** Let me select individual events.
    1. Check runs - _takes action, but only for the `getsentry` repo_
    2. Check suites - _not used afaict—don't remove here tho in order to avoid risk_
    3. Issues - _log metric_
    4. Issue comments - _log metric_
    5. Pull requests - _log metric_
    6. Pull request reviews - _log metric_
1. **Active:** [**not** checked]

### Confirming Current Repos

The webhook is currently installed directly on four repos.

```sql
select repository, count(*) n from open_source.github_events group by repository order by n desc;
```

| row | repository | n |
|---|---|---:|
| 1 | getsentry/sentry | 113731 |
| 2 | getsentry/sentry-docs | 8075 |
| 3 | getsentry/getsentry | 7840 |
| 4 | getsentry/onpremise | 5923 |

### Confirming Low Volume

We get < 1000 events per day during the week and < 100 per day on weekends.

```sql
select date_trunc(created_at, day) as day, count(*) n from open_source.github_events where date_trunc(created_at, day) > '2021-04-01' group by day order by day desc;
```

<img width="237" alt="Screen Shot 2021-05-25 at 11 45 52 AM" src="https://user-images.githubusercontent.com/134455/119527933-c713e280-bd4e-11eb-9166-1ab6576c079f.png">

### Confirming I Haz No Perms

I don't have permission to delete duplicates from `github_events`.

```sql
insert into open_source.github_events (repository) values ('garbage');
```

> Access Denied: Table super-big-data:open_source.github_events: Permission bigquery.tables.updateData denied on table super-big-data:open_source.github_events (or it may not exist). at [1:1]
